File: temp/fix_profile_upload_ui.py
from pathlib import Path

css_path = Path(r'NevariAdmin Storefront/app/globals.css')
dashboard_path = Path(r'NevariAdmin Storefront/app/_customer-dashboard.js')

css_lines = css_path.read_text(encoding='utf-8').splitlines()
for index, line in enumerate(css_lines):
    if line.strip() == '--spinner-size: 24px;':
        css_lines[index] = '  --spinner-size: 16px;'
        break
else:
    raise SystemExit('Spinner CSS line not found')
css_path.write_text('\n'.join(css_lines) + '\n', encoding='utf-8')

lines = dashboard_path.read_text(encoding='utf-8').splitlines()

def replace_block(start_line, end_line, new_block):
    try:
        start = lines.index(start_line)
    except ValueError as error:
        raise SystemExit(f'Start line not found: {start_line}') from error
    end = start
    while end < len(lines) and lines[end] != end_line:
        end += 1
    if end >= len(lines):
        raise SystemExit(f'End line not found after {start_line}: {end_line}')
    end += 1
    lines[start:end] = new_block

replace_block(
    '      const result = await uploadCustomerProfileImage(session, payload);',
    '      }',
    [
        '      const result = await uploadCustomerProfileImage(session, payload);',
        '      const nextAvatarUrl = result?.avatar_url || result?.src || "";',
        '      if (nextAvatarUrl) {',
        '        const nextProfile = {',
        '          ...(profile || {}),',
        '          ...(state.dashboard?.profile || {}),',
        '          avatar_url: nextAvatarUrl,',
        '          profile_image: nextAvatarUrl,',
        '        }',
        '        await mutateSummary((current) => current ? {',
        '          ...current,',
        '          profile: {',
        '            ...(current.profile || {}),',
        '            ...nextProfile,',
        '          },',
        '          dashboard: {',
        '            ...(current.dashboard || {}),',
        '            profile: {',
        '              ...(current.dashboard?.profile || {}),',
        '              ...nextProfile,',
        '            },',
        '          },',
        '        } : current, { revalidate: false })',
        '        setSession((current) => current ? {',
        '          ...current,',
        '          user: {',
        '            ...(current.user || {}),',
        '            avatar_url: nextAvatarUrl,',
        '            avatarUrl: nextAvatarUrl,',
        '            picture: nextAvatarUrl,',
        '          },',
        '        } : current)',
        '      }',
    ],
)

mobile_marker = "          swrKeys.proxy.path('/customer-dashboard/summary', withSessionCacheScope(session)),"
mobile_start = next((i for i, line in enumerate(lines) if line == '      const result = await uploadCustomerProfileImage(session, payload);' and i > 7000), None)
if mobile_start is None:
    raise SystemExit('Mobile start line not found')
mobile_end = mobile_start
while mobile_end < len(lines) and lines[mobile_end] != '      }':
    mobile_end += 1
if mobile_end >= len(lines):
    raise SystemExit('Mobile end line not found')
mobile_end += 1
lines[mobile_start:mobile_end] = [
    '      const result = await uploadCustomerProfileImage(session, payload);',
    '      const nextAvatarUrl = result?.avatar_url || result?.src || "";',
    '      if (nextAvatarUrl) {',
    '        const nextProfile = {',
    '          ...(profile || {}),',
    '          ...(state.dashboard?.profile || {}),',
    '          avatar_url: nextAvatarUrl,',
    '          profile_image: nextAvatarUrl,',
    '        }',
    '        await mobileGlobalMutate(',
    mobile_marker,
    '          (current) => current ? {',
    '            ...current,',
    '            profile: {',
    '              ...(current.profile || {}),',
    '              ...nextProfile,',
    '            },',
    '            dashboard: {',
    '              ...(current.dashboard || {}),',
    '              profile: {',
    '                ...(current.dashboard?.profile || {}),',
    '                ...nextProfile,',
    '              },',
    '            },',
    '          } : current,',
    '          { revalidate: false }',
    '        )',
    '        setSession((current) => current ? {',
    '          ...current,',
    '          user: {',
    '            ...(current.user || {}),',
    '            avatar_url: nextAvatarUrl,',
    '            avatarUrl: nextAvatarUrl,',
    '            picture: nextAvatarUrl,',
    '          },',
    '        } : current)',
    '      }',
]

dashboard_path.write_text('\n'.join(lines) + '\n', encoding='utf-8')
print('Applied profile upload fixes.')

File: temp/fix_profile_upload_ui.js
const fs = require('fs');

const dashboardPath = 'NevariAdmin Storefront/app/_customer-dashboard.js';
const cssPath = 'NevariAdmin Storefront/app/globals.css';

let dashboard = fs.readFileSync(dashboardPath, 'utf8');
let css = fs.readFileSync(cssPath, 'utf8');

const spinnerOld = '.customer-profile-upload-card .button-spinner {\r\n  --spinner-size: 24px;\r\n}';
const spinnerNew = '.customer-profile-upload-card .button-spinner {\r\n  --spinner-size: 16px;\r\n}';
if (!css.includes(spinnerOld)) {
  throw new Error('Spinner CSS block not found');
}
css = css.replace(spinnerOld, spinnerNew);

const desktopOld = '        const result = await uploadCustomerProfileImage(session, payload);\r\n' +
  '        const nextAvatarUrl = result?.avatar_url || result?.src || "";\r\n' +
  '        if (nextAvatarUrl) {\r\n' +
  '          await mutateSummary((current) => current ? {\r\n' +
  '            ...current,\r\n' +
  '            dashboard: {\r\n' +
  '              ...(current.dashboard || {}),\r\n' +
  '              profile: {\r\n' +
  '                ...(current.dashboard?.profile || profile || {}),\r\n' +
  '                avatar_url: nextAvatarUrl,\r\n' +
  '                profile_image: nextAvatarUrl,\r\n' +
  '              },\r\n' +
  '            },\r\n' +
  '          } : current, { revalidate: false });\r\n' +
  '        }\r\n';

const desktopNew = '        const result = await uploadCustomerProfileImage(session, payload);\r\n' +
  '        const nextAvatarUrl = result?.avatar_url || result?.src || "";\r\n' +
  '        if (nextAvatarUrl) {\r\n' +
  '          const nextProfile = {\r\n' +
  '            ...(profile || {}),\r\n' +
  '            ...(state.dashboard?.profile || {}),\r\n' +
  '            avatar_url: nextAvatarUrl,\r\n' +
  '            profile_image: nextAvatarUrl,\r\n' +
  '          };\r\n' +
  '          await mutateSummary((current) => current ? {\r\n' +
  '            ...current,\r\n' +
  '            profile: {\r\n' +
  '              ...(current.profile || {}),\r\n' +
  '              ...nextProfile,\r\n' +
  '            },\r\n' +
  '            dashboard: {\r\n' +
  '              ...(current.dashboard || {}),\r\n' +
  '              profile: {\r\n' +
  '                ...(current.dashboard?.profile || {}),\r\n' +
  '                ...nextProfile,\r\n' +
  '              },\r\n' +
  '            },\r\n' +
  '          } : current, { revalidate: false });\r\n' +
  '          setSession((current) => current ? {\r\n' +
  '            ...current,\r\n' +
  '            user: {\r\n' +
  '              ...(current.user || {}),\r\n' +
  '              avatar_url: nextAvatarUrl,\r\n' +
  '              avatarUrl: nextAvatarUrl,\r\n' +
  '              picture: nextAvatarUrl,\r\n' +
  '            },\r\n' +
  '          } : current);\r\n' +
  '        }\r\n';

if (!dashboard.includes(desktopOld)) {
  throw new Error('Desktop upload block not found');
}
dashboard = dashboard.replace(desktopOld, desktopNew);

const mobileOld = '      const result = await uploadCustomerProfileImage(session, payload);\r\n' +
  '      const nextAvatarUrl = result?.avatar_url || result?.src || "";\r\n' +
  '      if (nextAvatarUrl) {\r\n' +
  '        await mobileGlobalMutate(\r\n' +
  "          swrKeys.proxy.path('/customer-dashboard/summary', withSessionCacheScope(session)),\r\n" +
  '          (current) => current ? {\r\n' +
  '            ...current,\r\n' +
  '            dashboard: {\r\n' +
  '              ...(current.dashboard || {}),\r\n' +
  '              profile: {\r\n' +
  '                ...(current.dashboard?.profile || profile || {}),\r\n' +
  '                avatar_url: nextAvatarUrl,\r\n' +
  '                profile_image: nextAvatarUrl,\r\n' +
  '              },\r\n' +
  '            },\r\n' +
  '          } : current,\r\n' +
  '          { revalidate: false }\r\n' +
  '        );\r\n' +
  '      }\r\n';

const mobileNew = '      const result = await uploadCustomerProfileImage(session, payload);\r\n' +
  '      const nextAvatarUrl = result?.avatar_url || result?.src || "";\r\n' +
  '      if (nextAvatarUrl) {\r\n' +
  '        const nextProfile = {\r\n' +
  '          ...(profile || {}),\r\n' +
  '          ...(state.dashboard?.profile || {}),\r\n' +
  '          avatar_url: nextAvatarUrl,\r\n' +
  '          profile_image: nextAvatarUrl,\r\n' +
  '        };\r\n' +
  '        await mobileGlobalMutate(\r\n' +
  "          swrKeys.proxy.path('/customer-dashboard/summary', withSessionCacheScope(session)),\r\n" +
  '          (current) => current ? {\r\n' +
  '            ...current,\r\n' +
  '            profile: {\r\n' +
  '              ...(current.profile || {}),\r\n' +
  '              ...nextProfile,\r\n' +
  '            },\r\n' +
  '            dashboard: {\r\n' +
  '              ...(current.dashboard || {}),\r\n' +
  '              profile: {\r\n' +
  '                ...(current.dashboard?.profile || {}),\r\n' +
  '                ...nextProfile,\r\n' +
  '              },\r\n' +
  '            },\r\n' +
  '          } : current,\r\n' +
  '          { revalidate: false }\r\n' +
  '        );\r\n' +
  '        setSession((current) => current ? {\r\n' +
  '          ...current,\r\n' +
  '          user: {\r\n' +
  '            ...(current.user || {}),\r\n' +
  '            avatar_url: nextAvatarUrl,\r\n' +
  '            avatarUrl: nextAvatarUrl,\r\n' +
  '            picture: nextAvatarUrl,\r\n' +
  '          },\r\n' +
  '        } : current);\r\n' +
  '      }\r\n';

if (!dashboard.includes(mobileOld)) {
  throw new Error('Mobile upload block not found');
}
dashboard = dashboard.replace(mobileOld, mobileNew);

fs.writeFileSync(dashboardPath, dashboard);
fs.writeFileSync(cssPath, css);
console.log('Applied profile upload UI fixes.');
